src/net.cpp

```diff
@@ -1890,6 +1916,19 @@ void CConnman::DisconnectNodes()
                 // remove from m_nodes
                 m_nodes.erase(remove(m_nodes.begin(), m_nodes.end(), pnode), m_nodes.end());
 
+                // Add to reconnection list if appropriate. We don't reconnect right here, because
+                // the creation of a connection is a blocking operation (up to several seconds),
+                // and we don't want to hold up the socket handler thread for that long.
+                if (pnode->m_transport->ShouldReconnectV1()) {
+                    reconnections_to_add.push_back({
+                        .addr_connect = pnode->addr,
+                        .count_failure = pnode->m_count_failure_after_reconnect,
```

Comment:
6a2dbed1209b18d923da64b019f6f378d3c30695 I'm confused about the intention of `count_failure` / `m_count_failure_after_reconnect`. If this is set to `true` then `AddrManImpl` marks the most recent failed attempt and bumps `info.nAttempts`.

The latter suggests that we should set this to `false` if we'r trying a `v2` -> `v1` downgrade, otherwise we'd be counting double. Even better: we should not count the failed `v2` attempt if we intend to try again with `v1`.